### ISSUE TITLE
Doesn't activate

### ISSUE THREAD

**Issue by bottleworks** [flightaware/readonly-pi#1]
Hi.  I'm trying to get this going.  It installs just fine and executing "sudo switch-to-readonly -r" functions then reboots without error.  It doesn't appear to activate.  I run "touch /etc/this-is-a-test" and "touch /home/pi/this-is-a-test".  I then reboot.  Those should disappear after reboot, but they remain.  Looking through dmesg, I see "mmc0: host does not support reading read-only switch, assuming write-enable".  I'm not sure if that's related.  There doesn't appear to be anything else.  Any ideas?  What kicks this thing off at boot?  I've tried this on a Raspberry pi 0 W and a Raspberry pi 2.

**Comment by mutability** [comment]
Do you have an initramfs set up? The switch to readonly is done from within the initramfs

**Comment by bottleworks** [comment]
Thanks obj.  I overlooked that despite it clearly being noted.  I've never done anything with initramfs, so I'll try to figure that out.  (nor knew about it's existence until just now)  Thanks!

**Comment by bottleworks** [comment]
Just to dumb it down, it looks like I need to perform the actions noted at:  https://www.raspberrypi.org/forums/viewtopic.php?f=63&t=161416
Correct?
Or do I only need to do the following to make this setup function:
```
# uname -a
Linux raspberry 4.9.59+ #1047 Sun Oct 29 11:47:10 GMT 2017 armv6l GNU/Linux
# update-initramfs -c -k 4.9.59+
# cd /boot
# mv initrd.img-4.9.59+ initrd7.img
```

And the following in /boot/config.txt
```
kernel=kernel7.img
initramfs initrd7.img
```

and add "boot=overlay" in /boot/cmdline.txt

**Comment by mutability** [comment]
Just the latter. Note that you won't get automated rebuilds of initrd if you do it by hand like that. I think there is some automation for initrds in stretch, possibly (previously the rpi stuff did not integrate well with update-initramfs, but I think it is somewhat better now?)
